storage/Devices/MdContainerImpl.cc

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2017 SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact Novell, Inc.
+ *
+ * To contact Novell about this file by physical or electronic mail, you may
+ * find current contact information at www.novell.com.
+ */
+
+
+#include "storage/Devices/MdContainerImpl.h"
+#include "storage/Devices/MdMember.h"
+#include "storage/EtcMdadm.h"
+
+
+namespace storage
+{
+
+    using namespace std;
+
+
+    const char* DeviceTraits<MdContainer>::classname = "MdContainer";
+
+
+    MdContainer::Impl::Impl(const string& name)
+	: Md::Impl(name)
+    {
+	set_md_level(MdLevel::CONTAINER);
+    }
+
+
+    MdContainer::Impl::Impl(const xmlNode* node)
+	: Md::Impl(node)
+    {
+    }
+
+
+    void
+    MdContainer::Impl::check() const
+    {
+	Md::Impl::check();
+
+	if (get_md_level() != MdLevel::CONTAINER)
+	    ST_THROW(Exception("invalid md-level for MdContainer"));
+
+	if (get_metadata() != "" && get_metadata() != "imsm" && get_metadata() != "ddf")
```

Comment:
Not after probing. But MdContainer::create() keeps it empty - neither imsm nor ddf is a good default. So e.g. in testsuites it may be empty.